mtp_api/apps/prison/views.py

```diff
@@ -196,4 +196,35 @@ def form_valid(self, form):
             gettext('Deleted %(count)d balances.') % {'count': result['deleted']} + ' ' +
             gettext('Saved %(count)d balances.') % {'count': result['created']}
         ))
+        # prisoner balances are tied to prisoners by number AND prison
+        # either prisoner balances OR prisoner locations can be stale (depends on when files are generated and imported)
+        # so both need to link to prison and we should not automatically choose the prison being uploaded
+        chosen_prison = form.cleaned_data['prison']
+        # but since the wrong prison can be accidentally chosen, an error should show if this might have happened
+        likely_prison = self.prison_based_on_known_locations(chosen_prison)
+        if likely_prison and likely_prison != chosen_prison:
+            messages.error(self.request, gettext(
+                'File was uploaded for %(chosen_prison)s, '
+                'but database suggests that most prisoners are likely at %(likely_prison)s.'
+            ) % {
+                'chosen_prison': chosen_prison,
+                'likely_prison': likely_prison,
+            } + ' ' + gettext('Was the right prison selected?'))
```

Comment:
Do you reckon is too much to also suggest an action to correct? Or would it be too much text to read 🤷

Reply:
i didn't want it to _force_ someone to do something different. just being a bit cautious… even though i can't think of a scenario where it thinks you chose the wrong prison, but actually you submitted the right one. i suppose it could happen if a balance spreadsheet is really really small.